### PR TITLE
Per CPU map needed API and selftest

### PR DESCRIPTION
**Pull request by grantseltzer**

### CODE REVIEW

File: libbpfgo.go [hunk 1]
```diff
@@ -630,6 +638,15 @@ func (b *BPFMap) GetValue(key unsafe.Pointer) ([]byte, error) {
 	return value, nil
 }
 
+func (b *BPFMap) GetValue2(key unsafe.Pointer, value *[]byte) error {
```
Comment (kakkoyun) [inline libbpfgo.go:641]: Is it possible to have a better name than `GetValue2`? And also adding a documentation to explain the need of the additional method would be awesome!

Comment (grantseltzer) [reply libbpfgo.go:641]: Hahaha yes, I forgot that I pushed it with that name. I'll update it, thanks for your reviews @kakkoyun !

File: libbpfgo.go [hunk 2]
```diff
@@ -611,6 +611,14 @@ func (b *BPFMap) ValueSize() int {
 	return int(C.bpf_map__value_size(b.bpfMap))
 }
 
+func (b *BPFMap) SetValueSize(size uint32) error {
+	errC := C.bpf_map__set_value_size(b.bpfMap, C.uint(size))
```
Comment (javierhonduco) [inline libbpfgo.go:615]: Perhaps we could read both the return value and the error, as reading just one of the return values will result in us reading the C function return value, but not the errno

File: libbpfgo.go [hunk 3]
```diff
@@ -630,6 +638,15 @@ func (b *BPFMap) GetValue(key unsafe.Pointer) ([]byte, error) {
 	return value, nil
 }
 
+func (b *BPFMap) GetValue2(key unsafe.Pointer, value *[]byte) error {
+	valuePtr := unsafe.Pointer(&(*value)[0])
+	errC := C.bpf_map_lookup_elem(b.fd, key, valuePtr)
```
Comment (javierhonduco) [inline libbpfgo.go:643]: Same as above

**Comment by grantseltzer** [comment]
libbpf just merged [high level apis](https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/diff/tools/lib/bpf/libbpf.h?id=737d0646a83c) for accessing maps, which includes built in API for reading from per-cpu maps. This should be changed over to use that API (though it hasn't been released yet)

File: selftest/percpu/main.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+	if link.GetFd() == 0 {
+		os.Exit(-1)
+	}
+
+	go func() {
+		for {
+			syscall.Mmap(999, 999, 999, 1, 1)
```
Comment (rafaeldtinoco) [inline selftest/percpu/main.go:49]: This call is broken (like the other selftest with the same call). If you check for error here you will see this is not working (at least in the other exact same call for another PR).

File: libbpfgo.go (hunk 2 above)
Comment (grantseltzer) [reply libbpfgo.go:615]: bpf_map__set_value_size() returns the error directly and does not set errno.

**Review by rafaeldtinoco — APPROVED** [review]
LGTM. Make sure to squash needed commits and dont duplicate "Signed off by" lines in git log! Thanks!

**Pull request merged**